include/envoy/buffer/buffer.h

```diff
@@ -55,6 +56,20 @@ class BufferFragment {
   virtual void done() PURE;
 };
 
+/**
+ * A class to facilitate extracting buffer slices from a buffer instance.
+ */
+class SliceData {
+public:
+  virtual ~SliceData() = default;
+  /**
+   * @return absl::Span<uint8_t> a span of the slice data.
+   */
+  virtual absl::Span<uint8_t> getData() PURE;
```

Comment:
I think this should be something like absl::Span\<const uint8_t\>

I had not considered const issues in the existing buffer API which was a fairly direct translation of an earlier API built on top of libevent buffers.  IIRC data in Unowned slices should be considered immutable.  It is technically possible for multiple unowned slices to share the same underlying storage although I think we don't rely on that yet, but in-progress enhancements including the HTTP/1.1 cache extension would benefit from the ability to provide const access to data blocks that are referenced by buffers so multiple requests can reference the same data blocks in the in-memory cache.

Is use of const data blocks compatible with your use case?

Reply:
My use case requires a mutable slice, but I agree that the more common case would be an immutable view of the slice.  Commit df19673 attempts to provide both, with immutable as default.